src/main/java/hudson/remoting/Launcher.java

```diff
@@ -610,6 +613,7 @@ private SSLSocketFactory getSSLSocketFactory()
         return r;
     }
 
+    @SuppressFBWarnings(value = "XXE_DOCUMENT", justification = "This is the (so-called) JNLP file from the server, which has to be trusted from the perspective of this agent.")
```

Comment:
You can still disable external entities in the DBF/DB instances here.

Reply:
Good point. It doesn't hurt to add that, even though the server has to be trusted.